File: Ambulance.BLL/Models/CallModel.cs
﻿using NetTopologySuite.Geometries;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AmbulanceSystem.BLL.Models;

public class CallModel
{
    public int CallId { get; set; }
    public int PatientId { get; set; }
    public int DispatcherId { get; set; }
    public int? HospitalId { get; set; }
    public int? CallStatusId { get; set; }
    public string? Phone { get; set; }
    public int UrgencyType { get; set; }
    public Point? Address { get; set; }
    public DateTime? StartCallTime { get; set; }
    public DateTime? EndCallTime { get; set; }
    public DateTime? ArrivalTime { get; set; }
    public DateTime? CompletionTime { get; set; }
    public string? CallerName { get; set; }
    public string? PatientName { get; set; }
    public string? DispatcherName { get; set; }
    public string? HospitalName { get; set; }
    public string? Notes { get; set; }
    public TimeSpan EstimatedArrival { get; set; }
}

File: src/api/models/CallModel.js
import { BrigadeModel } from "./BrigadeModel.js";

export class CallModel {
    constructor({
        callId = 0,
        patientId = 0,
        dispatcherId = 0,
        hospitalId = null,
        callStatusId = null,
        phone = "",
        urgencyType = 0,
        address = "",
        assignedBrigades = [],
        startCallTime = null,
        endCallTime = null,
        arrivalTime = null,
        completionTime = null,
        callerName = "",
        patientName = "",
        dispatcherName = "",
        hospitalName = "",
        notes = "",
        estimatedArrival = "00:00:00"
    } = {}) {
        this.callId = callId;
        this.patientId = patientId;
        this.dispatcherId = dispatcherId;
        this.hospitalId = hospitalId;
        this.callStatusId = callStatusId;
        this.phone = phone;
        this.urgencyType = urgencyType;
        this.address = address;
        this.assignedBrigades = assignedBrigades.map(b => new BrigadeModel(b));
        this.startCallTime = startCallTime ? new Date(startCallTime) : null;
        this.endCallTime = endCallTime ? new Date(endCallTime) : null;
        this.arrivalTime = arrivalTime ? new Date(arrivalTime) : null;
        this.completionTime = completionTime ? new Date(completionTime) : null;
        this.callerName = callerName;
        this.patientName = patientName;
        this.dispatcherName = dispatcherName;
        this.hospitalName = hospitalName;
        this.notes = notes;
        this.estimatedArrival = estimatedArrival; // з TimeSpan можна передавати як рядок
    }
}